src/main.rs

```diff
@@ -484,34 +535,119 @@ impl BFInterp {
         }
     }
 
-    fn sext(&mut self, op: &Operand, to_type: &TypeRef, dest: &name::Name) {
+    fn szext(&mut self, op: &Operand, to_type: &TypeRef, dest: &name::Name) {
```

Comment:
Any reason we renamed this to `szext` in this PR?

Reply:
Yes I'm using the same function for both sext and zext instructions.

Reply:
Although it might be a bit confusing.

Reply:
I see. I'll leave it up to you then what you want to name it.

Reply:
I'll keep name.